### PR TITLE
Improve CPE Helper with new update field

### LINKED ISSUE #5295: Add new entries for the `update` field in the CPE Helper
Hi team,

This issue is to adapt the documentation to the latest changes in the Vulnerability Detector:
- https://github.com/wazuh/wazuh/issues/8434

These changes should be reflected in the documentation as follows:

- [x] Include in the "**_CPE Helper Schema_**" section the new optional field `update`, both for source and translation.
- [x] Add the two new options: `set_update_if_product_matches` and `set_update_only_if_product_matches`. In the "**_Translation types_**" section.
- [x] It would also be good to include a complete section explaining how the new entry of the "_Microsoft Exchange Server_" package is generated.

### PR DESCRIPTION
**Pull request by MarcelKemp**
## Description

|Related issue|
|---|
|closes #5295|

After adding the update field in the CPE Helper, it has been necessary to update the Wazuh documentation to include the new options, variables and entries, reflecting the new functionalities.

To do so, the following points have been implemented:

- [x] Include in the "**_CPE Helper Schema_**" section the new optional field `update`, both for source and translation.
- [x] Add the two new options: `set_update_if_product_matches` and `set_update_only_if_product_matches`. In the "**_Translation types_**" section.
- [x] Adding a section reflecting the example of the software product "Microsoft Exchange Server", where the added functionalities are used (translation with the use of the `update` field, use of the variable `$UPDATE` extracted from the regex using the action `set_update_if_product_match` and example of a vulnerability _CVE-2022-23277_ affected by the package).

## Checks
- [x] It compiles without warnings.
- [x] Spelling and grammar. 
- [x] Used impersonal speech. 
- [x] Used uppercase only on nouns.

### CODE REVIEW

File: source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst [hunk 1]
```diff
@@ -586,3 +598,207 @@ To illustrate the process, we will follow the same procedure as in the previous
...
+If we consult this product in the `National Vulnerability Database (CVE-2022-23277) <https://nvd.nist.gov/vuln/detail/CVE-2022-23277>`_,
+we will find the following CPEs:
+
+
+.. thumbnail:: ../../../images/manual/vuln-detector/nvd-exchange-server.png
+    :title: Affected software for CVE-2022-23277
+    :align: center
+    :width: 100%
+
+Where we can observe, that in the CPEs the version and update fields are specified with generic versions that do not allow us
```
Comment (chemamartinez) [inline source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst:618]: I'd try to reorganize this sentence to make it sound less colloquial. In addition, I don't understand the `generic versions` meaning, they seem specific versions.

File: source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst [hunk 2]
```diff
@@ -586,3 +598,207 @@ To illustrate the process, we will follow the same procedure as in the previous
...
+to know the patch to fix the vulnerability.
+
+To know the patches that would mitigate the vulnerability, it is necessary to consult the `Microsoft Security Update Guide <https://msrc.microsoft.com/update-guide/vulnerability/CVE-2022-23277>`_,
+where we can know the patches that fix the vulnerability for the *Microsoft Exchange Server* product.
```
Comment (chemamartinez) [inline source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst:622]:
```suggestion
where we can find the patches that fix the vulnerability for the *Microsoft Exchange Server* product.
```

File: source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst [hunk 3]
```diff
@@ -586,3 +598,207 @@ To illustrate the process, we will follow the same procedure as in the previous
...
+
+.. thumbnail:: ../../../images/manual/vuln-detector/msug-exchange-server.png
+    :title: CVE-2022-23277 in the Microsoft Security Update Guide
+    :align: center
+    :width: 100%
+
+As in the previous use case, we will activate the ``check_hotfix`` action to automate this search,
```
Comment (chemamartinez) [inline source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst:629]:
```suggestion
As explained in the previous use case, we have to include the ``check_hotfix`` action to automate this search,
```

File: source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst [hunk 4]
```diff
@@ -586,3 +598,207 @@ To illustrate the process, we will follow the same procedure as in the previous
...
+in addition to following the same guide used for the other use cases explained.
```
Comment (chemamartinez) [inline source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst:630]:
```suggestion
as well as following the same guide used for the previous use case.
```

File: source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst [hunk 5]
```diff
@@ -586,3 +598,207 @@ To illustrate the process, we will follow the same procedure as in the previous
...
+
+  To extract the CPEs from the program, we can use the information previously consulted
+  on the CPEs affected by *CVE-2022-23277*. From that source, we can see exactly which CPEs
+  are checked to confirm the vulnerability. These are:
+
+  - ``cpe:2.3:a:microsoft:exchange_server:2016:cumulative_update_21:*:*:*:*:*:*``
+  - ``cpe:2.3:a:microsoft:exchange_server:2016:cumulative_update_22:*:*:*:*:*:*`` (our package would match this CPE)
+
+  We can see that the two target programs only differ in the field *update* (cumulative_update_*nº*). So we need to use
+  the ``set_update_if_product_match`` option to add the *update* field to differentiate between the CPEs, as there will be
```
Comment (chemamartinez) [inline source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst:682]: three `to` in the same sentence is difficult to read, let's try something like:
```
In this case, we need to add the ``set_update_if_product_match`` action so the *update* field is included in the CPE to match the one provided by the NVD

File: source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst [hunk 6]
```diff
@@ -586,3 +598,207 @@ To illustrate the process, we will follow the same procedure as in the previous
...
+            "replace_vendor",
+            "replace_product",
+            "set_version_if_product_matches",
+            "set_update_if_product_matches",
+            "replace_msu_name",
+            "check_hotfix"
+        ]
+    }
+
+  Alert example using this dictionary entry:
```
Comment (chemamartinez) [inline source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst:755]:
```suggestion
Once applied the entry in the CPE helper and the scanner detects it, its vulnerabilities are reported:
```

File: source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst [hunk 7]
```diff
@@ -586,3 +598,207 @@ To illustrate the process, we will follow the same procedure as in the previous
...
+  | Update              |                                                      | ^Microsoft Exchange Server.*Update ([0-9]+)                      | cumulative_update_$(UPDATE) = cumulative_update_22  | set_update_if_product_matches        |
+  +---------------------+------------------------------------------------------+------------------------------------------------------------------+-----------------------------------------------------+--------------------------------------+
+  | MSU name            |                                                      | Microsoft Exchange Server $(VERSION) Cumulative Update $(UPDATE) | Microsoft Exchange Server 2016 Cumulative Update 22 | replace_msu_name                     |
+  +---------------------+------------------------------------------------------+------------------------------------------------------------------+-----------------------------------------------------+--------------------------------------+
+
+  The resulting dictionary entry, including more versions of *Microsoft Exchange Server*:
```
Comment (chemamartinez) [inline source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst:709]:
```suggestion
Here we have the CPE helper entry needed to detect vulnerabilities of the *Microsoft Exchange Server* program:
```

**Review by chemamartinez — CHANGES_REQUESTED** [review]

**Review by chemamartinez — APPROVED** [review]

**Review by chemamartinez — APPROVED** [review]

**Comment by javimed** [comment]
Further work specified in https://github.com/wazuh/wazuh-documentation/issues/5581

**Review by chemamartinez — APPROVED** [review]
LGTM!

**Review by javimed — APPROVED** [review]

**Pull request merged**